scripts/release.sh

```diff
@@ -136,6 +136,41 @@ function setup_branch() {
   git fetch ${KNATIVE_UPSTREAM} ${RELEASE_BRANCH}:upstream/${RELEASE_BRANCH}
 }
 
+# Setup version, branch and release notes for a auto release.
+function prepare_auto_release() {
+  echo "Auto release requested"
+  TAG_RELEASE=1
+  PUBLISH_RELEASE=1
+
+  local tags="$(git tag | cut -d 'v' -f2 | cut -d '.' -f1-2 | sort | uniq )"
+  local branches="$( { (git branch -r | grep origin/release-) ; (git branch  | grep release-); } | cut -d '-' -f2 | sort | uniq)"
+  local last_version=""
+
```

Comment:
For safety & debugging sake, check that `tags` and `branches` are not empty. Also, initialize `RELEASE_VERSION` to empty string.